packages/web-react/src/components/Grid/useGridItemStyleProps.ts

```diff
@@ -0,0 +1,59 @@
+import { ElementType, CSSProperties } from 'react';
+import { useClassNamePrefix } from '../../hooks';
+import { GridColumn, GridColumnBreakpoints, SpiritGridItemProps } from '../../types';
+
+interface GridItemCSSProperties extends CSSProperties {
+  [key: string]: string | undefined | number;
+}
+
+const setStyleProperty = (styleObject: GridItemCSSProperties, propertyName: string, value: string | undefined) => {
+  (styleObject as Record<string, string | undefined>)[propertyName] = value;
+};
+
+const setGridItemStyle = (styleObject: GridItemCSSProperties, baseVarName: string, propValue: GridColumn) => {
+  if (typeof propValue === 'number') {
+    const propName = `--${baseVarName}`;
+    setStyleProperty(styleObject, propName, propValue.toString());
+  } else if (typeof propValue === 'object' && propValue !== null) {
+    Object.keys(propValue).forEach((key) => {
+      const suffix = key === 'mobile' ? '' : `-${key}`;
+      const propName = `--${baseVarName}${suffix}`;
+      setStyleProperty(styleObject, propName, propValue[key as keyof GridColumnBreakpoints]?.toString());
+    });
+  }
+};
+
+export interface GridItemStyles<T> {
+  /** className props */
+  classProps: string;
+  /** Props for the grid element. */
+  props: T;
+  /** Style props for the grid element */
+  styleProps: GridItemCSSProperties;
+}
+
+export function useGridItemStyleProps(
+  props: SpiritGridItemProps<ElementType>,
+): GridItemStyles<SpiritGridItemProps<ElementType>> {
+  const { columnEnd, columnSpan, columnStart, ...restProps } = props;
+
+  const gridItemClass = useClassNamePrefix('GridItem');
+
+  const gridItemStyle: GridItemCSSProperties = {};
+
+  if (columnEnd) {
+    setGridItemStyle(gridItemStyle, 'grid-item-column-end', columnEnd);
```

Comment:
I am not a big fan of passing and modifying values by reference. I like simple I/O functions that are testable by the way they return something based on input.

Also I think that the modification of the grid item style can be done by iterating over the props and maybe do some clever `map` function. Just quick refactor:

```
const columnPropNames = Object.keys(props).filter((propName) => propName.startsWith('column'));
  columnPropNames.forEach((propName) => {
    if (props[propName]) {
      setGridItemStyle(
        gridItemStyle,
        `grid-item-column-${propName.replace('column', '').toLowerCase()}`,
        props[propName],
      );
    }
  });
```

Or something similar should be more general. But your solution is also working 🤷